src/Features/Core/Portable/Diagnostics/Service/DiagnosticAnalyzerService_GetDiagnosticsForSpan.cs

```diff
@@ -224,33 +248,18 @@ bool TryDeprioritizeAnalyzer(
             //  2. We are processing 'CodeActionRequestPriority.Normal' priority request.
             //  3. Analyzer registers certain actions that are known to lead to high performance impact due to its broad analysis scope,
             //     such as SymbolStart/End actions and SemanticModel actions.
-            //  4. Analyzer did not report a diagnostic on the same line in prior document snapshot.
 
             // Conditions 1. and 2.
             if (kind != AnalysisKind.Semantic ||
                 !span.HasValue ||
-                priorityProvider.Priority != CodeActionRequestPriority.Default)
+                priority != CodeActionRequestPriority.Default)
             {
                 return false;
             }
 
-            Debug.Assert(span.Value.Length < text.Length);
-
             // Condition 3.
             // Check if this is a candidate analyzer that can be de-prioritized into a lower priority bucket based on registered actions.
-            if (!deprioritizationCandidates.Contains(analyzer))
-                return false;
-
-            // 'LightbulbSkipExecutingDeprioritizedAnalyzers' option determines if we want to execute this analyzer
-            // in low priority bucket or skip it completely. If the option is not set, track the de-prioritized
-            // analyzer to be executed in low priority bucket.
-            // Note that 'AddDeprioritizedAnalyzerWithLowPriority' call below mutates the state in the provider to
-            // track this analyzer. This ensures that when the owner of this provider calls us back to execute
-            // the low priority bucket, we can still get back to this analyzer and execute it that time.
-            if (!this._globalOptions.GetOption(DiagnosticOptionsStorage.LightbulbSkipExecutingDeprioritizedAnalyzers))
-                priorityProvider.AddDeprioritizedAnalyzerWithLowPriority(analyzer);
-
-            return true;
+            return await this.IsDeprioritizedAnalyzerAsync(project, analyzer, cancellationToken).ConfigureAwait(false);
```

Comment:
this is the major change.  instead of trying to dynamically change things as we run analyzers.  we just compute this information per analyzer and check for it here.